### ISSUE TITLE
Remaining Gradio 3.23 issues

### ISSUE THREAD

**Issue by AlUlkesh** [AlUlkesh/stable-diffusion-webui-images-browser#100]
There currently are still several issues, where the new Gradio version causes weird interactions.

Most of these should be fixed, when the current PRs on the webui's side go through.

I think it would be a waste of time to hunt and fix something until then. Hope that goes faster than PRs did recently.

**Comment by mart-hill** [comment]
For me, I totally don't see any "send" buttons other, than "folder" one. 🙂

**Comment by AlUlkesh** [comment]
That's a bit odd though. It mostly works for me. Do you have the latest commits of both webui and this extension?

Also it seems Firefox is having more touble than Chrome based browsers.

**Comment by mart-hill** [comment]
> That's a bit odd though. It mostly works for me. Do you have the latest commits of both webui and this extension?
> 
> Also it seems Firefox is having more touble than Chrome based browsers.

Yes, both of them are the lastest. "main"/"master" branches. I'm using Edge.
By the way, I still included `if v:` fix in `generation_parameters_copypaste.py` file, but now it doesn't matter, because there are no buttons to use. 🙂 
By the way, SendTo buttons in **PNG Info** tab also don't seem to be working - they just switch the tab, and that's it.

**Comment by mart-hill** [comment]
For me, it looks like this:
![image](https://user-images.githubusercontent.com/55754730/227816299-63e89728-d0cb-4119-b4cb-eafda8967435.png)

 I use Adguard as a system-wide ad-filtering software (with their network driver), but "localhost" is excluded from filtering (by me).

**Comment by AlUlkesh** [comment]
Can you hit F12 and see if there are any messages in the javascript console about image browser? There are currently many about other parts, those you can ignore at the moment.

**Comment by AlUlkesh** [comment]
One other thing, I've noticed the gallery size seems to be a factor. Does changing it to the default 6 x 6 do anything?

**Comment by mart-hill** [comment]
Drat, I forced return to WebUI's 64b7e8382377bb578d9740c061979776b214cfd9 commit to be actually able to use it at all... When I started your current version of the exntension, I saw the buttons "appearing back" under this old gradio 3.16.2, and AdGuard doesn't stand in the way at all. :)
![image](https://user-images.githubusercontent.com/55754730/227849833-8344ef39-04b3-4241-b1ee-bf33f28ca8ed.png)



I didn't touch the gallery "size" setting, I think, but I'll check that! It's 6x6, though. 
![image](https://user-images.githubusercontent.com/55754730/227849613-2fc50458-77d6-44d0-b410-caf1efe5cb20.png)

**Comment by forsurefr** [comment]
After pulling the latest commit, these issues still exist:
- The checkpoint list has a loading animation when this extension is active so you cannot change the checkpoint: https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/9034
- The paging system in the image browser does not work properly, you cannot reach the latest image and the thumbnails are too small, the larger thumbnails (100 per page) no longer appear.

**Comment by AlUlkesh** [comment]
@forsurefr 

These Gradio issues really have me scratching my head. I don't mind bugs caused by it or changes I have to make, as long as I can test them. But this Gradio update seems to have left different issues with different people. For example the checkpoint list issue you describe is not happenening for me.

Frustrating.

Anyway, can you hit F12 and see if there are any messages in the javascript console about image browser?

**Comment by TheLastCreation** [comment]
I'm adding this here instead of a new issue since it only came about after pulling the latest in both git repo's
at first I was getting an error related to line 178, accordion.classList.contains("rotate-90")) being null, adding in a check if null my next error isn't so easy for me debug.

image_browser.js?1679937079.8286097:183 
 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'classList')
    at image_browser_controlnet_send (image_browser.js?167…7079.8286097:183:13)
image_browser_controlnet_send	@	image_browser.js?1679937079.8286097:183
await in image_browser_controlnet_send (async)		
image_browser_controlnet_send_txt2img	@	image_browser.js?1679937079.8286097:199
eval	@	VM586:3
ze	@	Blocks.svelte:283
(anonymous)	@	Blocks.svelte:342
(anonymous)	@	index.37b24c07.js:4
zr	@	index.37b24c07.js:4
_	@	Button.svelte:12
(anonymous)	@	index.37b24c07.js:4
zr	@	index.37b24c07.js:4
c	@	Button.svelte:11

**Comment by Puddz** [comment]
> @forsurefr
> 
> These Gradio issues really have me scratching my head. I don't mind bugs caused by it or changes I have to make, as long as I can test them. But this Gradio update seems to have left different issues with different people. For example the checkpoint list issue you describe is not happenening for me.
> 
> Frustrating.
> 
> Anyway, can you hit F12 and see if there are any messages in the javascript console about image browser?

Hey, I'm having the same kind of issue. When I have this extension enabled, I get a loading symbol for everything basically. All my quick settings at the top and every setting in the settings tab.

This is what shows in the console. 
![image](https://user-images.githubusercontent.com/1914381/228022797-d2dad606-5441-401e-8868-ec0371ac07d1.png)

**Comment by AlUlkesh** [comment]
I just found a couple of issue's with some of webui's javascripts. If noone has done this yet, I'll put in PR.

**Comment by AlUlkesh** [comment]
Can someone pull my webui PR and check if it also improves some image browser issues?
https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/9060

**Comment by forsurefr** [comment]
> @forsurefr
> 
> These Gradio issues really have me scratching my head. I don't mind bugs caused by it or changes I have to make, as long as I can test them. But this Gradio update seems to have left different issues with different people. For example the checkpoint list issue you describe is not happenening for me.
> 
> Frustrating.
> 
> Anyway, can you hit F12 and see if there are any messages in the javascript console about image browser?

@AlUlkesh
This is the only error in the console, when interacting with the gallery pages:
```
Blocks.svelte:214 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'forEach')
    at ct (Blocks.svelte:214:8)
    at Blocks.svelte:293:7

**Comment by Puddz** [comment]
I also have what @forsurefr has in my console as well. Disabled deforum and looked and that was there.

**Comment by AlUlkesh** [comment]
That's been there before the Gradio changes, shouldn't be an issue, For more details see #28.

**Comment by AlUlkesh** [comment]
@forsurefr
Can you also go to Settings and check "Image Browser - Print debug logs to the console". Then restart webui and post the complete log, so starting from:
```
c:\tools\Ai\stable-diffusion-webui>webui-user.bat
venv "c:\tools\Ai\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)]
Commit hash: 955df7751eef11bb7697e2d77f6b8a6226b21e13
```
Perhaps we can find another hint there.

**Comment by AlUlkesh** [comment]
Which OS, web browser, python version are you using?

**Comment by wmooring** [comment]
So, I found out on my fresh SD install that I originally did to fix an issue with the preview images not loading on the main tab, that if the "https://github.com/AlUlkesh/stable-diffusion-webui-images-browser.git" extension is enabled, the Checkpoint dropdown at the top and the contents Settings tab will never load, the "spinning orange boxes" loading icons just stay.  Disabling the extension, restarting the UI, and then reloading the browser tab will allow the settings to show up again.  The tab reload doesn't help as long as the extension is enabled.

However, I found out this issue affects desktop Firefox but *not* desktop Chrome.  Mobile Chrome and Firefox also have the issue.  Which is a shame, because I kind of need the extension to use SD from my phone, since I can't browse my desktop folders from it effectively.

Image Browser extension is version a9032d4d  (Mon Mar 27 14:14:06 2023)

The status bar of the bottom of my UI:
python: 3.10.6  •  torch: 1.13.1+cu117  •  xformers: N/A  •  gradio: 3.23.0  •  commit: 955df775  •  checkpoint: 6ce0161689

**Comment by forsurefr** [comment]
> So, I found out on my fresh SD install that I originally did to fix an issue with the preview images not loading on the main tab, that if the "https://github.com/AlUlkesh/stable-diffusion-webui-images-browser.git" extension is enabled, the Checkpoint dropdown at the top and the contents Settings tab will never load, the "spinning orange boxes" loading icons just stay. Disabling the extension, restarting the UI, and then reloading the browser tab will allow the settings to show up again. The tab reload doesn't help as long as the extension is enabled.
> 
> However, I found out this issue affects desktop Firefox but _not_ desktop Chrome. Mobile Chrome and Firefox also have the issue. Which is a shame, because I kind of need the extension to use SD from my phone, since I can't browse my desktop folders from it effectively.
> 
> Windows 10 Pro and an Android phone, Image Browser extension is version [a9032d4](https://github.com/AlUlkesh/stable-diffusion-webui-images-browser/commit/a9032d4d32b6e005ba47ed8b5438b61337cb6330) (Mon Mar 27 14:14:06 2023)
> 
> The status bar of the bottom of my UI: python: 3.10.6  •  torch: 1.13.1+cu117  •  xformers: N/A  •  gradio: 3.23.0  •  commit: 955df775  •  checkpoint: 6ce0161689

My setup is the same but using Brave on desktop. Have not tried Chrome or Firefox.

**Comment by AlUlkesh** [comment]
I just tried Chrome, Firefox and Brave, but they all do not have this issue for me (Win 10).

I suggest you try the PR I quoted above, I think that solves a lot of issues.

**Comment by AlUlkesh** [comment]
This seems to have helped some people:
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/9055#issuecomment-1485760587

**Comment by ItsLogic** [comment]
> This seems to have helped some people: [AUTOMATIC1111/stable-diffusion-webui#9055 (comment)](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/9055#issuecomment-1485760587)

I can confirm this fixed my infinite settings loading with firefox on linux. Just added it to the bottom of style.css and everything loads as normal on 656ef38d
Webui info:
`python: 3.10.10  •  torch: 2.0.0+cu118  •  xformers: N/A  •  gradio: 3.23.0  •  commit: 955df775`

**Comment by AlUlkesh** [comment]
This commit fixes the issue where you would get out of the visible window, when moving left or right with the small thumbnails at the bottom.

**Comment by forsurefr** [comment]
> This seems to have helped some people: [AUTOMATIC1111/stable-diffusion-webui#9055 (comment)](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/9055#issuecomment-1485760587)

I can confirm that this worked for me. Now the only thing that is broken is the paging in image browser. After I click first or next page, the image browser stops responding and a loading animation appears, and loading indefinitely.

**Comment by AlUlkesh** [comment]
I saw some issue describing that webui is doing "something" for 20 - 30 seconds upon startup, where something like you describe can happen. Just to test, can you wait 30 seconds and then see if it perhaps works?

**Comment by AlUlkesh** [comment]
@forsurefr 
Did you try all the suggestions here: #107 ?

**Comment by EllangoK** [comment]
Thanks for the fixes so far, but just curious does anyone's arrow key's work and update the sidebar and delete pointers?

It changes the image for me, but basically the pointer doesn't change so like on the side bar the generation info doesn't update, and when deleting it will delete the wrong image. This is in both direction as well, just after restarting in both Chrome and Firefox, with just this and control net. 

I have looked at the fixes above as well and applied them, but they didn't help. The debug info print out shows that yeah it doesn't change when you use the arrow key to navigate images, but it still does when you click on the thumbnails on the bottom row,

I will see if I can look into it myself, but I just want to make sure it wasn't just me as I thought someone else would have reported this already.

**Comment by EllangoK** [comment]
I have found what will probably give us our workaround if this a real problem, https://gradio.app/docs/#gallery-select-header.

The select event both runs on clicking, and when you navigate with arrow keys, so we can just probably change the sidebar based on this event.

I'll probably write a fix soon, but I would like someone to confirm if this is a real problem.

**Comment by EllangoK** [comment]
@AlUlkesh This is off a fresh install, only having ever generated these images and it still does not work for me, in both firefox and chrome. Really don't know what would be causing it in this scenario.

![arrow-keys-not-working](https://user-images.githubusercontent.com/23466035/228838503-a33ff4d5-b855-4ff9-9c90-9c2e3385eb71.gif)

https://github.com/EllangoK/stable-diffusion-webui-images-browser/commit/ac54ba03e9d136e0a05221f61593b29572b56c8f does fix it for me, maybe you might want to look at it? 

I just duplicated show_image_info temporarily just to show the difference. It also removes the need for the _js I believe so it just might be better overall.

**Comment by AlUlkesh** [comment]
@EllangoK 
I've given up trying to understand why the Gradio changes have such vastly different effects for different people...

I tried your code and using the new select method is certainly nicer.

But it seems to get out of sync, once you hit the right key fast:

https://user-images.githubusercontent.com/99896447/228847780-ad477d8a-1a22-4e22-a887-970cb7ad9be7.mp4

**Comment by EllangoK** [comment]
@AlUlkesh yeah gradio really is a pain, and I'm not blaming you ofc man, you've done a great job maintaining probably what is one of the most annoying extensions to maintain.

I didn't realise it would desync but yeah it does, I'll see if I can do anything about that. I feel like gradio should have ensured that doesn't desync as it is a method they provide, but like they really give us no guarantees about anything.

**Comment by AlUlkesh** [comment]
@EllangoK 
Thanks 😄 

This is from the Gradio docs:
`On the Gradio server, a queue is set up, which adds each request that comes to a list. When a worker is free, the first available request is passed into the worker for inference. When the inference is complete, the queue sends the prediction back through the websocket to the particular Gradio user who called that prediction.`

I suspect the "workers" work in parallel, so the order of the values returned to python is kinda random, if one is slower than another?

**Comment by horribleCodes** [comment]
"Send to ControlNet" still doesn't work for me. I can't tell when it stopped working, but I just pulled 88388c7 and there's nothing happening when I click the button. I also have debug and warning logs enabled, and there's nothing in the terminal either.

**Comment by AlUlkesh** [comment]
@horribleCodes 
I'll look into it.

As for when it stopped working, "with the Gradio changes" is always the answer...

**Comment by AlUlkesh** [comment]
@horribleCodes, @onewoodmot
My testing has shown, that with the Gradio update I'm no longer able to consistently target the necessary controlnet UI elements.

So I just put in a PR on the controlnet side.

Once that goes through I can commit my updates.

I have it locally running again.

**Comment by MrKuenning** [comment]
Send to controlnet also broke for the openpose extension for a few days after the gradio update. But it is now working again. Maybe you could see how they are doing it.   Thanks again for your hard work and how responsive you are with the community.

**Comment by AlUlkesh** [comment]
Yes, I saw that. But their solution always sends to Controlnet 0.

**Comment by AlUlkesh** [comment]
**all:** Can you please check if this change solved any of your issues: #126?

About controlnet: Apparently the maintainer has been absent. Hope he comes back.

**Comment by MrKuenning** [comment]
I am fine with just having ControlNet 0 for now.  Most of the time I only have a single layer anyway.

**Comment by AlUlkesh** [comment]
@MrKuenning 
Yeah, I hadn't expected the merge to take this long. I'll get a at-least-controlnet-0 version ready.

**Comment by AlUlkesh** [comment]
I actually managed to get all controlnets working... 😃

**Comment by MrKuenning** [comment]
You Rock!

**Comment by horribleCodes** [comment]
It looks like the "Send to" buttons are missing when using the image browser from a mobile device.

**Comment by MrKuenning** [comment]
![Screenshot_20230408-021844.png](https://user-images.githubusercontent.com/16549268/230716027-e89654e2-a243-4a67-ae22-9df04d201799.png)

**Comment by AlUlkesh** [comment]
@horribleCodes 
Works on my android. Anything on the javascript console?

**Comment by GeorgLegato** [comment]
Hi, I fixed for my extension (VectorStudio) the sendtoControlnet today.
The official codebase as of today was still not compat with gradio 3.23

here the code snippet for the function "image_browser_controlnet_send"
(skipping the data-img stuff....)
```

	const container = gradioApp().querySelector(toTab === "txt2img" ? "#txt2img_script_container" : "#img2img_script_container")

	const accordion = container.querySelector("#controlnet .label-wrap")
	if (!accordion.classList.contains("open")) accordion.click()  // on first time no DOM there!
	
	const tab = gradioApp().querySelectorAll("#tab_"+toTab+" #controlnet .tab-nav button")[controlnetNum]
	if (!tab) {
		// come back after click() has created the DOM 
		setTimeout(()=>{
			vectorstudio_controlnet_send(toTab,controlnetNum)
		},1000)
		return
	}

	if (!tab.classList.contains("selected")) tab.click()

	const input = gradioApp().querySelectorAll("#tab_"+toTab+" #controlnet input[type='file']")[controlnetNum]
	/*  try {
			a click seems not to be neccessary 
			input.parentElement.click()
		} catch (e) {}
	*/

	input.value = ""
	input.files = dt.files
	input.dispatchEvent(new Event("change", { bubbles: true, composed: true }))

	// switch to txt2/img2img and scroll to controlnet-tab
	vectorstudio_gototab(toTab, "tabs", tab)

```

on initial load the click()-method have a race condition if not per design a glitch. the click on tab does create the DOM, which is done defered. So I add a timeout after click() and come back to the same routine try to handle it, now.

replace "vectorstudio_" to imagebrowser of course.

**Comment by horribleCodes** [comment]
> @horribleCodes Works on my android. Anything on the javascript console?

Honestly at this point nothing about the extension is doing anything on my tablet. I'm not getting any log messages at all, even with all logging enabled - it's working fine on my PC.
The WebUI feels much slower and less responsive on my tablet overall, so I'm not sure what exactly is going on - but I'm definitely noticing the lack of any form of response from the image browser.

**Comment by AlUlkesh** [comment]
@horribleCodes 
That is truly odd. If you enabled the image browser debug logging, you should at the very least get the initial logs: https://github.com/AlUlkesh/stable-diffusion-webui-images-browser/blob/9e72ee78f434b073d7103cc1782118de83e0388d/scripts/image_browser.py#L151

This code is executed at the very beginning, before any ui is even created.

If you don't even get that, that would probably mean, the script is not being loaded (correctly) by webui. But then that should be on your console.

I'm a little lost here.

**Comment by AlUlkesh** [comment]
Can anyone who still has issues please follow the new debug info guide: #130 and post your logs?

**Comment by horribleCodes** [comment]
> Can anyone who still has issues please follow the new debug info guide: #130 and post your logs?

Okay, even at log level 3 my tablet isn't triggering any log messages, even though it works fine on my PC. However, that was on Brave, and when I tried 2 different Chromium browsers, the images loaded. However, one refused to load the EXIF data, and both still aren't showing the "Send To"-buttons. Here's what I get when I loaded the first page and selected an image:

```
2023-04-10-22:49:24 image_browser.py: get_image_page
2023-04-10-22:49:24 image_browser.py: get_all_images
2023-04-10-22:49:24 image_browser.py: curr_path: ...\stable-diffusion-webui\outputs\txt2img-images
2023-04-10-22:49:24 image_browser.py: cache_exif: 41/5100 cache_aes: 41/5100 0.2 seconds
2023-04-10-22:49:24 image_browser.py: get_image_thumbnail
2023-04-10-22:51:51 image_browser.py: show_image_info: tab_base_tag_box, num, page_index, len(filenames), num_of_imgs_per_page: image_browser_tab_txt2img, 23, 1.0, 5100, 36
2023-04-10-22:55:56 image_browser.py: get_image_page
2023-04-10-22:55:56 image_browser.py: get_all_images
2023-04-10-22:55:56 image_browser.py: curr_path: ...\stable-diffusion-webui\outputs\txt2img-images
2023-04-10-22:55:56 image_browser.py: cache_exif: 0/5100 cache_aes: 0/5100 0.0 seconds
2023-04-10-22:55:56 image_browser.py: get_image_thumbnail
2023-04-10-22:56:05 image_browser.py: show_image_info: tab_base_tag_box, num, page_index, len(filenames), num_of_imgs_per_page: image_browser_tab_txt2img, 23, 1.0, 5100, 36
2023-04-10-22:56:21 image_browser.py: show_image_info: tab_base_tag_box, num, page_index, len(filenames), num_of_imgs_per_page: image_browser_tab_txt2img, 24, 1.0, 5100, 36
```

**Comment by AlUlkesh** [comment]
Can you all try the latest update please: #135

**Comment by Kapowpenguin1** [comment]
unsure if this is a bug or just a feature that's yet to be migrated from prior to 3.23, but i've found that
1. I can no longer zoom into images. I can view the gallery, and click on an image to open it, but i used to be able to click on it again to open a larger view. I can no longer do that. instead, it simply advances to the next image
2. image metadata is no longer being displayed. prior to 3.23, clicking on an image would load all its metadata into the boxes on the right. now, the boxes instead stay empty

if these are indeed bugs, i will submit debug logs

**Comment by AlUlkesh** [comment]
The first issue is a webui issue. I actually have submitted a PR to fix that. But it's been waiting for weeks to get merged by now. Check the third point here: #107

You might want to install that manually and try it out.

**Comment by Kapowpenguin1** [comment]
> The first issue is a webui issue. I actually have submitted a PR to fix that. But it's been waiting for weeks to get merged by now. Check the third point here: #107
> 
> You might want to install that manually and try it out.

how would i do a manual install? am i supposed to git clone your repo into my webui directory?

**Comment by Kapowpenguin1** [comment]
> There are several ways to do it. The cleanest would probably be to create a local branch like this:
> 
> ```
> git fetch origin pull/9060/head:pr9060
> git checkout pr9060
> ```
> 
> If you want to switch back to the master branch later, do:
> 
> `git checkout master`

thanks- 9060 was successfully able to fix the zoom issue. ig i just wait for it to be merged, now

**Comment by Blargswell** [comment]
> unsure if this is a bug or just a feature that's yet to be migrated from prior to 3.23, but i've found that
> 
> 1. I can no longer zoom into images. I can view the gallery, and click on an image to open it, but i used to be able to click on it again to open a larger view. I can no longer do that. instead, it simply advances to the next image
> 2. image metadata is no longer being displayed. prior to 3.23, clicking on an image would load all its metadata into the boxes on the right. now, the boxes instead stay empty
> 
> if these are indeed bugs, i will submit debug logs
> 
> edit: nevermind...? ig i'm just on an outdated version... i installed image browser on a fresh install of the webui and everything works perfectly. now to... figure out how to get the webui to acknowledge the latest version edit 2: i'm very confused. the fresh install doesn't face either problem, but the updated image browser [i just ran `git pull` in the extension directory] in my original install still faces problem 1...

I was having the same problem, but thanks to someone else bringing it up, the issue was being caused by my Open Pose extension (for some reason) being stuck on an outdated version. Every time I tried to update it from the UI, it would just stay out of date. I had to delete it and reinstall it fresh to get it up to date, but once that was done, all of my problems in the image browser went away.

**Comment by Kapowpenguin1** [comment]
> > unsure if this is a bug or just a feature that's yet to be migrated from prior to 3.23, but i've found that
> > 
> > 1. I can no longer zoom into images. I can view the gallery, and click on an image to open it, but i used to be able to click on it again to open a larger view. I can no longer do that. instead, it simply advances to the next image
> > 2. image metadata is no longer being displayed. prior to 3.23, clicking on an image would load all its metadata into the boxes on the right. now, the boxes instead stay empty
> > 
> > if these are indeed bugs, i will submit debug logs
> > edit: nevermind...? ig i'm just on an outdated version... i installed image browser on a fresh install of the webui and everything works perfectly. now to... figure out how to get the webui to acknowledge the latest version edit 2: i'm very confused. the fresh install doesn't face either problem, but the updated image browser [i just ran `git pull` in the extension directory] in my original install still faces problem 1...
> 
> I was having the same problem, but thanks to someone else bringing it up, the issue was being caused by my Open Pose extension (for some reason) being stuck on an outdated version. Every time I tried to update it from the UI, it would just stay out of date. I had to delete it and reinstall it fresh to get it up to date, but once that was done, all of my problems in the image browser went away.

so i did what you mentioned, switched back to the master branch, then ran `git pull` in the openpose-editor directory, and restarted the webui. and voila, image zoom and descriptions work properly now. very interesting

**Comment by GeorgLegato** [comment]
> > > unsure if this is a bug or just a feature that's yet to be migrated from prior to 3.23, but i've found that
> > > 
> > > 1. I can no longer zoom into images. I can view the gallery, and click on an image to open it, but i used to be able to click on it again to open a larger view. I can no longer do that. instead, it simply advances to the next image
> > > 2. image metadata is no longer being displayed. prior to 3.23, clicking on an image would load all its metadata into the boxes on the right. now, the boxes instead stay empty
> > > 
> > > if these are indeed bugs, i will submit debug logs
> > > edit: nevermind...? ig i'm just on an outdated version... i installed image browser on a fresh install of the webui and everything works perfectly. now to... figure out how to get the webui to acknowledge the latest version edit 2: i'm very confused. the fresh install doesn't face either problem, but the updated image browser [i just ran `git pull` in the extension directory] in my original install still faces problem 1...
> > 
> > 
> > I was having the same problem, but thanks to someone else bringing it up, the issue was being caused by my Open Pose extension (for some reason) being stuck on an outdated version. Every time I tried to update it from the UI, it would just stay out of date. I had to delete it and reinstall it fresh to get it up to date, but once that was done, all of my problems in the image browser went away.
> 
> so i did what you mentioned, switched back to the master branch, then ran `git pull` in the openpose-editor directory, and restarted the webui. and voila, image zoom and descriptions work properly now. very interesting

I do not switch from any other branch to main or so, my experience is the bug is sporadic: sometimes i can open a small preview in details view (zoom view) and sometimes not.

**Comment by Kapowpenguin1** [comment]
sure, i just mention that i switched from the master branch bc i previously was on pr9060 to fix the issue

**Comment by GeorgLegato** [comment]
found something about the "Cant see zoomed image" bug:
I try to open 10 times an image.. no success, no zommed winows opened.
In that moment I open the debug dev-tools in my Edge, I can open any image and zoomed view.
Is there a timing/sync problem, which is impacted by debugging, slower execution time due to dev-tools on?

**Comment by AlUlkesh** [comment]
It's possible. I also have had another bug where the issue goes away if you make your browser window smaller, which would also be true in your case with devtools taking up a lot of the space.

Have you tried the PR I mentioned above to see if the problem goes away?

**Comment by JoeyLearnsToCode** [comment]
I don’t know if it’s related to this issue. The front-end performance of the image browser is particularly poor in the latest version of webui (of course, the front-end performance of the latest version of webui is not good). I know it’s related to Gradio. Almost every operation requires waiting for Gradio’s animation. In the image browser, I like to click and operate very quickly. Now I can’t use it smoothly.

不知道和这个issue有没有关系，image browser 在最新版本的 webui 中前端性能格外差（当然，最新版本 webui 的前端性能本来也不好），我知道和 gradio 有关系，几乎每个操作后都要等 gradio 的动画，在 image browser 中我喜欢很迅速地点击、操作，现在我不能很流畅地使用了。

**Comment by AlUlkesh** [comment]
@JoeyLearnsToCode 
When I first switched to the new Gradio I also noticed a massive slowdown. However, in my case, I realized it was because my system RAM was filling out and it started swapping out to the SSD. Once I got more RAM it was fine.

So perhaps, you can check, how your system's status is, when those slowdowns happen?

**Comment by JoeyLearnsToCode** [comment]
@AlUlkesh thank you for reply, my RAM capacity is 32GB, that should not be the reason in my case.
For example, when I use left/right key switch image, two loading animations show above the image, that is where "Current ranking" and "Set ranking to" locate. The animations exist 0.2~0.3s, it's not too long, but I just can't stand it, maybe I'm being picky.
In the old version of webui, there is no loading animation. Now, feels like these animations are keep telling me, "you are using a remote service by internet, instead of a local web app", but I am using a local web app! :(

**Comment by AlUlkesh** [comment]
@JoeyLearnsToCode 
So you think it's more a perceived delay than an actual one? Hmm... I just added the option to turn the progress indicator off. Check if this feels better for you.
![image](https://user-images.githubusercontent.com/99896447/236632833-e85fe462-237a-4799-adce-9579d438d4b5.png)

**Comment by JoeyLearnsToCode** [comment]
@AlUlkesh wow, much better!
But now I find another problem, the "Generation Info" part update with 1s delay after switching image(which is realtime and fast).
Within that second of delay, delete button seems to delete the previous image(before switch), not the new one(after).

**Comment by AlUlkesh** [comment]
@JoeyLearnsToCode Let's look at this issue here: #159
